### ISSUE TITLE
interactive retrieval demo

### ISSUE THREAD

**Issue by MXueguang** [castorini/pyserini#548]
We have our first example of demo https://github.com/castorini/pyserini/blob/master/pyserini/demo/msmarco.py
as described in https://github.com/castorini/pyserini/pull/546.
Currently, it is a demo with Sparse search only, we want it to support Dense search and Hybrid search as well.
So based on current code, we want to add commands like `/model ance` or `/model tct` and `/mode hybrid` etc to config search models and type.
Basically, integrate the example code [here](https://github.com/castorini/pyserini#how-do-i-search) into the demo feature.

**Comment by saileshnankani** [comment]
I'm currently working on this issue

**Comment by MXueguang** [comment]
@lintool we need wikipedia demos too, right?

**Comment by lintool** [comment]
with MCP, this is defunct now... #2135
